contracts/actions/mcd/McdRepayComposite.sol

```diff
@@ -0,0 +1,268 @@
+            _returnValues
+        );
+
+        repayParams.exchangeData.srcAddr = _parseParamAddr(
+            repayParams.exchangeData.srcAddr,
+            _paramMapping[4],
+            _subData,
+            _returnValues
+        );
+        repayParams.exchangeData.destAddr = _parseParamAddr(
+            repayParams.exchangeData.destAddr,
+            _paramMapping[5],
+            _subData,
+            _returnValues
+        );
+
+        repayParams.exchangeData.srcAmount = _parseParamUint(
+            repayParams.exchangeData.srcAmount,
+            _paramMapping[6],
+            _subData,
+            _returnValues
+        );
+
+        _flBalancer(repayParams);
+    }
+
+    /// @notice Parses inputs and runs the single implemented action through a proxy
+    /// @dev Used to save gas when executing a single action directly
+    function executeActionDirect(bytes memory _callData) public payable virtual override (ActionBase, DFSSell) {
+        RepayParams memory repayParams = _parseCompositParams(_callData);
+        _flBalancer(repayParams);
+    }
+
+    /// @notice Gets a FL from Balancer and returns back the execution to the action address
+    function _flBalancer(RepayParams memory _repayParams) internal {
+        address[] memory tokens = new address[](1);
+        uint256[] memory amounts = new uint256[](1);
+        tokens[0] = _repayParams.exchangeData.srcAddr;
+        amounts[0] = _repayParams.repayAmount;
+
+        IManager(_repayParams.mcdManager).cdpAllow(_repayParams.vaultId, ACTION_ADDR, 1);
+
+        IFlashLoans(VAULT_ADDR).flashLoan(
+            ACTION_ADDR,
+            tokens,
+            amounts,
+            abi.encode(_repayParams, address(this))
+        );
+
+        IManager(_repayParams.mcdManager).cdpAllow(_repayParams.vaultId, ACTION_ADDR, 0);
+    }
+
+    /// @notice Balancer FL callback function that formats and calls back this Action contract
+    function receiveFlashLoan(
+        address[] memory _tokens,
+        uint256[] memory _amounts,
+        uint256[] memory _feeAmounts,
+        bytes memory _userData
+    ) external override nonReentrant {
+        require(msg.sender == VAULT_ADDR, "Untrusted lender");
+        (RepayParams memory repayParams, address proxy) = abi.decode(_userData, (RepayParams, address));
+
+        _repay(proxy, repayParams);
+
+        uint256 flPaybackAmount = _amounts[0] + _feeAmounts[0];
+        _tokens[0].withdrawTokens(address(VAULT_ADDR), flPaybackAmount);
+    }
+
+    function _repay(address _proxy, RepayParams memory _repayParams) internal {
+        uint256 collateral = getAllColl(IManager(_repayParams.mcdManager), _repayParams.joinAddr, _repayParams.vaultId);
+        _repayParams.repayAmount = _repayParams.repayAmount > collateral ? collateral : _repayParams.repayAmount;
+
+        // Sell flashloaned collateral asset for debt asset
+        assert(_repayParams.exchangeData.srcAmount == _repayParams.repayAmount); // CONSIDER
```

Comment:
makes sense, but sending different srcAmount and repayAmount should never happen, what about setting srcAmount to repayAmount?

Reply:
removed unneeded repayAmount param, srcAmount used instead